### PR TITLE
ISPN-16569 Bloom filter doesn't track updates properly

### PR DESCRIPTION
**Pull request by wburns**
https://issues.redhat.com/browse/ISPN-16569

### CODE REVIEW

File: client/hotrod-client/src/main/java/org/infinispan/client/hotrod/near/NearCacheService.java [hunk 1]
```diff
@@ -183,6 +197,7 @@ public MetadataValue<V> get(K key) {
    @Override
    public void clear() {
       cache.clear();
+      nearCacheRemovals.set(0);
```
Comment (jabolina) [inline client/hotrod-client/src/main/java/org/infinispan/client/hotrod/near/NearCacheService.java:200]: Should it have a null check here?

Comment (jabolina) [reply client/hotrod-client/src/main/java/org/infinispan/client/hotrod/near/NearCacheService.java:200]: Ah yeah, seems the tests are hitting it.

**Comment by wburns** [comment]
Yes, good catch. I had it in my other PR and forgot to copy it back over.

**Comment by wburns** [comment]
Updated

**Review by jabolina — APPROVED** [review]

**Comment by rigazilla** [comment]
@wburns the jira number in the commit msg is not correct

**Comment by wburns** [comment]
Fixed JIRA number, thanks!

**Comment by tristantarrant** [comment]
Several thread leaks around `ClusterInvalidatedNearCacheBloomTest.ThreadLeakChecker `

**Comment by wburns** [comment]
I am closing as this will be fixed in the new client and bloom filter has been broken for so long, I don't know if it is worth it here.

**Pull request closed without merging**